torchgeo/datamodules/reforestree.py

```diff
@@ -0,0 +1,72 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""ReforesTree datamodule."""
+
+from typing import Any
+
+from torch.utils.data import Subset
+import kornia.augmentation as K
+
+from ..datasets import ReforesTree
+from .geo import NonGeoDataModule
+from .utils import group_shuffle_split
+
+
+class ReforesTreeDataModule(NonGeoDataModule):
+    """LightningDataModule implementation for the ReforesTree dataset.
+
+    Implements 80/20 train/val splits.
+
+    .. versionchanged:: 0.1
+        
+    """
+
+    def __init__(
+        self, batch_size: int = 64, num_workers: int = 0, size: int = 512, **kwargs: Any
+    ) -> None:
+        """Initialize a new ReforesTreeDataModule instance.
+
+        Args:
+            batch_size: Size of each mini-batch.
+            num_workers: Number of workers for parallel data loading.
+            size: resize images of input size 512x512 to size x size
```

Comment:
The images are 4,000 x 4,000 px, not 500 x 500 px, right? For images this large, you may want to consider using `_RandomNCrop`. See `OSCDDataModule` for an example of this.